src/main/scala/com/mwz/sonar/scala/scoverage/ScoverageMetrics.scala

```diff
@@ -0,0 +1,94 @@
+package com.mwz.sonar.scala.scoverage
+
+import org.sonar.api.measures.{CoreMetrics, Metric, Metrics}
+import scala.collection.JavaConverters._
+
+/**
+ * Statement coverage metric definition.
+ *
+ * @author Rado Buransky
+ * @author BalmungSan
+ */
+class ScoverageMetrics extends Metrics {
+  override def getMetrics: java.util.List[Metric[_ <: java.io.Serializable]] =
+    List[Metric[_ <: java.io.Serializable]](
+      ScoverageMetrics.totalStatements,
+      ScoverageMetrics.coveredStatements,
+      ScoverageMetrics.statementCoverage,
+      ScoverageMetrics.branchCoverage
+    ).asJava
+}
+
+object ScoverageMetrics {
+
+  /** Builds a new [[Metric]] */
+  private def BuildMetric[T <: java.io.Serializable](
+    metricKey: String,
+    metricName: String,
+    metricType: Metric.ValueType,
+    metricDescription: String,
+    metricDirection: java.lang.Integer,
+    metricDomain: String,
+    isMetricQualitative: Boolean           = false,
+    metricValues: Option[(Double, Double)] = None
+  ): Metric[T] = {
+    val metricBuilder = new Metric.Builder(metricKey, metricName, metricType)
+      .setDescription(metricDescription)
+      .setDirection(metricDirection)
+      .setDomain(metricDomain)
+      .setQualitative(isMetricQualitative)
+
+    metricValues match {
+      case Some((worstValue, bestValue)) =>
+        metricBuilder
+          .setWorstValue(worstValue)
+          .setBestValue(bestValue)
+          .create[T]()
+      case None => metricBuilder.create[T]()
+    }
+  }
+
+  val totalStatements: Metric[java.lang.Integer] =
+    BuildMetric(
+      metricKey = "total_statements",
```

Comment:
I'm wondering whether we should align those custom scoverage metric keys with those built into Sonar e.g. [see here](https://docs.sonarqube.org/display/SONAR/Metric+Definitions#MetricDefinitions-Tests).
So according to the docs we could use the following:

- `statements`  instead of `total_statements`
- `scoverage` - is there an existing Sonar metric key for this?
- `branch_coverage` instead of `branch_scoverage`?

Assuming that we use those built-in metrics, then we probably don't need to be creating all of these custom ones and instead, we can just save new measures directly on the context when we need to during the analysis.

Does this make sense?

Reply:
Ok, that's fair enough.